feat(curriculum): update handleclick method to use state function

suburban-daredevil opened the pull request:
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] The handleClick method is now updated to get the user to use the updater function inside the setState method instead of directly mutating the state of the component

Closes #42208

Updated both the seed code and the answer code to enforce the usage of updater function inside the setState method. This is done in order to maintain the consistency of data inside the state of the component.

suburban-daredevil committed:
Updated the handleClick method to use updater function in react tutorial english

lasjorg commented:
For the incomplete updater function do we want the empty function body, or not? I'm just thinking about the use of an implicit return and having a starting syntax that might be more "open" to it.

So this:
```
this.setState(prevState => prevState);
```

Instead of this:

```
this.setState(prevState => {

});
```

I'm fine with it either way really, so LGTM.

suburban-daredevil commented:
Hello @lasjorg. Thanks for going with my pull request. 

Personally as a learner myself, I think having two `prevState` keywords might confuse the learner who is a beginner while learning about the updater function in `this.setState` in React.

So it is my humble suggestion to go with this :

```
this.setState(prevState => {

});
```
However if you want to me to change the code, I am more than happy to do the same. Thanks once again for selecting my PR.

ShaunSHamilton commented on curriculum/challenges/english/03-front-end-libraries/react/render-conditionally-from-props.md:
```diff
@@ -240,8 +240,9 @@ class GameOfChance extends React.Component {
   handleClick() {
-    this.setState({
-      counter: 0 // Change this line
+    // Complete the setState updater function
+    this.setState(prevState => {
+
```
```suggestion
    this.setState(prevState => {
      // Complete the return statement:
      return {
        counter: prevState
      }
```

ShaunSHamilton reviewed (changes requested):
Hey there,

We are only using a set of [agreed-upon comments](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/dictionaries/english/comments.json), and should not be adding to them (for i18n reasons).

suburban-daredevil committed:
Update comments in the seed code of curriculum/challenges/english/03-front-end-libraries/react/render-conditionally-from-props.md

Changing comments in the seed code according to the list of [agreed-upon comments](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/dictionaries/english/comments.json)

Co-authored-by: Shaun Hamilton <shauhami020@gmail.com>

suburban-daredevil commented:
Hello @ShaunSHamilton,

I have committed the changes that were requested. Thank you!

ShaunSHamilton reviewed (approved):
This LGTM 👍 . I am happy with this. Thank you, for making that change.

@lasjorg Feel free to agree/disagree

lasjorg reviewed (approved):
@ShaunSHamilton Didn't know about that. I think the comment was just something I added to the initial code suggestion.

LGTM
